SiameseDataset.py

```diff
@@ -0,0 +1,271 @@
+from __future__ import print_function
+import collections
+import h5py
+import numpy.random as random
+import numpy as np
+
+from CachedDataset2 import CachedDataset2
+from HDFDataset import FeatureSequenceStreamParser, SparseStreamParser, SegmentAlignmentStreamParser
+from Dataset import DatasetSeq
+
+
+
+class SiameseHDFDataset(CachedDataset2):
+  """
+  SiameseHDFDataset class allows to do sequence sampling for weakly-supervised training
+  It accepts data in the format of NextGenHDFDataset and performs sampling of sequence triplets before each epoch
+  Here we assume that each dataset from the input .hdf has a single label
+  Sampled triplets are splitted into different streams
+  In the config we can access streams by e.g. ["data:features_0", "data:features_1", "data:features_2"],
```

Comment:
So the dataset provides exactly those 3 input data streams, always named `features_0`, `features_1` and `features_2`? Or does it depend somehow on the parameter `input_stream_name` or so?
And then also exactly always 3 output data streams, named `classes_0`, `classes_1` and `classes_2`?
This should be very clear from the documentation.